tests/testCommon/file_server.go

```diff
@@ -29,9 +31,30 @@ func StartStaticFileServer(t *testing.T) *http.Server {
 			t.Fail()
 		}
 	}()
+
+	err := WaitForServerReady(10)
+	if err != nil {
+		t.Fail()
+	}
+
 	return server
 }
 
+// WaitForServerReady waits for server ready until timeout
+func WaitForServerReady(timeoutInSeconds uint) error {
```

Comment:
After some quick research, apparently the right thing to do is not use ListenAndServe at all.
http://grokbase.com/t/gg/golang-nuts/141ee4dqyg/go-nuts-how-to-know-when-listenandserve-is-ready-to-handle-connections
Would you like to modify with this solution?

Reply:
Thought about it, current solution based on client pulling also works for ListenAndServeTLS if we want to test https in the future.